archive/mod.ts

```diff
@@ -0,0 +1,40 @@
+// Copyright 2018-2022 the Deno authors. All rights reserved. MIT license.
+
+/*!
+ * Ported and modified from: https://github.com/beatgammit/tar-js and
+ * licensed as:
+ *
+ * (The MIT License)
+ *
+ * Copyright (c) 2011 T. Jameson Little
+ * Copyright (c) 2019 Jun Kato
+ * Copyright (c) 2018-2022 the Deno authors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+/**
+ * Provides a `Tar` and `Untar` classes for compressing and decompressing
+ * arbitrary data.
+ *
+ * @module
+ */
+/** @deprecated (will be removed after 0.169.0) TODO: export all once overlapping objects/types are removed */
+export { Tar, type TarData, type TarDataWithSource } from "./tar.ts";
+export * from "./untar.ts";
```

Comment:
`mod.ts` doesn't exist now. Does it make sense to add these deprecated versions here?

Reply:
Previously, `tar.ts` acted as the module entry point. Now that `Tar` and `Untar` classes have been separated, `mod.ts` acts as the entry point for both classes.

Reply:
It'll also provide a module description where [one currently doesn't exist for `std/archive`](https://deno.land/std?doc).